elastalert/queries/elasticsearch_query.py

```diff
@@ -428,18 +431,22 @@ def remove_duplicates(self, data: List[dict]) -> List[dict]:
 
 class ElasticsearchAggregationQuery(ElasticsearchQuery):
     def run(self, endtime: datetime) -> Tuple[datetime, datetime, int]:
-        starttime = self.set_starttime(endtime)
+        if self.rule_config.get("initial_starttime"):
```

Comment:
This could be a problem in combination with the spike metric aggregation rule because the rule relies on calling set_starttime.

Reply:
The initial behavior before refactoring the sourcecode (query-handling) always selected the initial_starttime as start-time, if it was set. This way a starttime can be set through commandline arguments. This was defined independent on ruletype. Removing the initial_starttime from aggregationquery leads to no side-effects for the maasaggregationrule, as the run method which handles the logic is overwritten anyaways. So this is easily possible if setting the starttime to initial_starttime leads to unwanted side-effects.

For me the question is: should the initial_starttime be considered (set by cmd-arg) when executing the rule? If not, I will revert the change in ElasticsearchAggregationQuery.run

Reply:
I've thought it through. This should be no problem. Only if you specify a problem which is not enough in the past, the result may be some false positives at the beginning because some of the reference windows are empty.